### DISCUSSION TITLE
How to cleanup old data?

### DISCUSSION THREAD

**Discussion by kostya9** [grafana/tempo#1487] [Q&A] [answered]
For example, imagine using local disk as the storage backend. Can I limit the data by time (e.g. store data for a month only) or limit data by storage (e.g. store only 1 TB of data and delete oldest entries if the limit is reached)?

**Comment by mapno** [comment, marked as answer]
Hi! Yes, the compactor is the component tasked with handling data retention. Retention in Tempo is defined based on time. To configure it, use the [`compactor.block_retention`](https://grafana.com/docs/tempo/latest/configuration/#compactor) config option. It defaults to 14 days. Once that a block is older than the configured retention period, the compactor will delete it.

**Comment by joe-elliott** [comment]
I would also recommend setting a lifecycle policy on your object storage that deletes objects a day or two after your Tempo retention. If a compactor or ingester OOMs while creating a block it can orphan objects and Tempo will lose track of them. This should be rare, but a lifecycle policy will prevent these objects from existing forever.
